wfe2/wfe.go

```diff
@@ -2377,6 +2378,71 @@ func (wfe *WebFrontEndImpl) RenewalInfo(ctx context.Context, logEvent *web.Reque
 		time.Unix(0, cert.Expires).UTC()))
 }
 
+// UpdateRenewal is used by the client to inform the server that they have
+// replaced the certificate in question, so it can be safely revoked. All
+// requests must be authenticated to the account which ordered the cert.
+func (wfe *WebFrontEndImpl) UpdateRenewal(ctx context.Context, logEvent *web.RequestEvent, response http.ResponseWriter, request *http.Request) {
+	if !features.Enabled(features.ServeRenewalInfo) {
+		wfe.sendError(response, logEvent, probs.NotFound("Feature not enabled"), nil)
+		return
+	}
+
+	body, _, acct, prob := wfe.validPOSTForAccount(request, ctx, logEvent)
+	addRequesterHeader(response, logEvent.Requester)
+	if prob != nil {
+		// validPOSTForAccount handles its own setting of logEvent.Errors
+		wfe.sendError(response, logEvent, prob, nil)
+		return
+	}
+
+	var updateRenewalRequest struct {
+		CertID   string `json:"certID"`
+		Replaced bool   `json:"replaced"`
+	}
+	err := json.Unmarshal(body, &updateRenewalRequest)
+	if err != nil {
+		wfe.sendError(response, logEvent, probs.Malformed("Unable to unmarshal RenewalInfo POST request body"), err)
+		return
+	}
+
+	der, err := base64.RawURLEncoding.DecodeString(updateRenewalRequest.CertID)
+	if err != nil {
+		wfe.sendError(response, logEvent, probs.Malformed("certID was not base64url-encoded"), nil)
+		return
+	}
+
+	var id certID
+	rest, err := asn1.Unmarshal(der, &id)
+	if err != nil || len(rest) != 0 {
+		wfe.sendError(response, logEvent, probs.Malformed("certID was not a DER-encoded CertID sequence"), nil)
+		return
+	}
+
+	if !id.HashAlgorithm.Algorithm.Equal(asn1.ObjectIdentifier{2, 16, 840, 1, 101, 3, 4, 2, 1}) {
+		wfe.sendError(response, logEvent, probs.Malformed("certID used hash algorithm other than SHA-256"), nil)
+		return
+	}
+
```

Comment:
```suggestion
	// We can do all of our processing based just on the serial, because Boulder
	// does not re-use the same serial across multiple issuers.
```